code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -214,6 +284,30 @@ obj/item/weapon/gun/energy/laser/retro
 	projectile_type = /obj/item/projectile/energy/plasma
 	charge_cost = 50
 
+/obj/item/weapon/gun/energy/plasma/failure_check(var/mob/living/carbon/human/M)
+	if(damaged && prob(25))
+		fire_delay += rand(6, 12)
+		var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+		s.set_up(3, 1, src)
+		s.start()
+		to_chat(M, "<span class='warning'>The [name] sparks violently.</span>")
+		return 1
+	if(damaged && prob(10))
+		M.drop_item()
+		M.emote("scream",,, 1)
+		M.adjustFireLossByPart(10, LIMB_LEFT_HAND, src)
+		M.adjustFireLossByPart(10, LIMB_RIGHT_HAND, src)
+		to_chat(M, "<span class='danger'>The [name] burns your hands!.</span>")
+		return 0
+	if(damaged && prob(1+fire_delay/3))
+		var/turf/T = get_turf(loc)
+		explosion(T, 0, 1, 3, 5)
+		M.drop_item(src, force_drop = 1)
+		qdel(src)
+		to_chat(M, "<span class='danger'>The [name] explodes!.</span>")
+		return 0
+	..()
```

Comment:
When I end failure_checks with the ..(), it prevents guns affected by them from firing and I don't know why. ..() should call the "earlier" versions of the proc, right? So this would call obj/item/weapon/gun/energy/failure_check, then that would call obj/item/weapon/gun/failure_check, which should always return 1 and let the gun fire if it makes it there. And if it hits one of the effects before then, they should return something and let it shoot or not, but it never does. The ricochet rifle's fire() effects still work with the ..() after them, so I don't know why failure_check doesn't. The assault rifle will still jam, so it's going past gun/projectile/automatic/failure to gun/projectile/failure, but it never seems to get back to gun/failure